packages/react-reconciler/src/ReactFiberCommitWork.old.js

```diff
@@ -2880,7 +2880,7 @@ function commitMutationEffectsOnFiber(
         }
 
         if (isHidden) {
-          if (!wasHidden) {
+          if (current !== null && current.memoizedState === null) {
```

Comment:
Nice find! Maybe add a comment explaining why this check is different from `wasHidden`? Or rename/refactor `wasHidden` somehow